pkg/reconciler/reconciler.go

```diff
@@ -375,8 +376,33 @@ func WithPostHook(h hook.PostHook) Option {
 	}
 }
 
+// WithValueTranslator is an Option that configures a function that translates a
+// custom resource to the values passed to Helm.
+// Use this if you need to customize the logic that translates your custom resource to Helm values.
+// If you wish to, you can convert the Unstructured that is passed to your Translator to your own
+// Custom Resource struct with code leveraging the k8s.io/apimachinery/pkg/runtime converter:
+//
+//	  foo := your.Foo{}
+//	  if err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, &foo); err != nil {
+//	    return nil, err
+//	  }
+//    // work with the type-safe foo
+//
+// Alternatively, your translator can also work similarly to a Mapper, by accessing the spec with:
+//
+//    u.Object["spec"].(map[string]interface{})
+func WithValueTranslator(t values.Translator) Option {
+	return func(r *Reconciler) error {
+		r.valueTranslator = t
+		return nil
+	}
+}
+
 // WithValueMapper is an Option that configures a function that maps values
-// from a custom resource spec to the values passed to Helm
+// from a custom resource spec to the values passed to Helm.
+// Use this if you want to apply a transformation on the values obtained from your custom resource, before
+// they are passed to Helm.
+// This functionality will be removed in a future release. Use WithValueTranslator instead.
```

Comment:
Start a separate paragraph with the prefix `Deprecated: `

From a Go [blog post about deprecations](https://go.dev/blog/godoc):
>  Sometimes a struct field, function, type, or even a whole package becomes redundant or unnecessary, but must be kept for compatibility with existing programs. To signal that an identifier should not be used, add a paragraph to its doc comment that begins with “Deprecated:” followed by some information about the deprecation.

Linting tools look for this and provide helpful warnings to developers using deprecated code.

```suggestion
//
// Deprecated: This functionality will be removed in a future release. Use WithValueTranslator instead.
```

Reply:
Done.